src/Comps/search/recentSearches/RecentSearches.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {
+  Dispatch,
+  MouseEventHandler,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+import { Container, Row, Title, ClearButton, Search } from "./style";
+import {} from "../../../styles/layouts";
+import {} from "../../../styles/typography";
+import { DeleteIcon } from "../../../Icons";
+
+type ISearch = {
+  value: string[];
+  setInput: Dispatch<SetStateAction<string>>;
+};
+const RecentSearches: React.FC<ISearch> = ({ value, setInput }) => {
+  const [items, setItems] = useState<string[]>(value);
+
+  const clearSearches = (event: { preventDefault: () => void }) => {
+    event.preventDefault();
+    localStorage.setItem("lastSearches", JSON.stringify([]));
+    setItems([]);
+  };
+  const chooseSearch = (item: string) => {
+    setInput(item);
+  };
+  const deleteSearch = (item: string,event: { preventDefault: () => void; }) => {
+    console.log(item);
+    event.preventDefault();
+
+  };
+  useEffect(() => {
+    if (items) {
+      const item = localStorage.getItem("lastSearches");
```

Comment:
you have this line of code also in another file.
Think how you can maybe create your own generic useLocalStorage custom hook that you would be able to each time you need to use your local storage. not only for that recentSearches use case